File: algorithms/0118.pascals_triangle/pascals_triangle.rs
impl Solution {
    fn generate(num_rows: i32) -> Vec<Vec<i32>> {
        let num_rows = num_rows as usize;
        let mut triangle = vec![vec![1]];

        for n in 1..num_rows {
            let last_line = &triangle[n - 1];
            let mut line = vec![1];

            for i in 0..last_line.len() - 1 {
                let sum = last_line[i] + last_line[i + 1];
                line.push(sum);
            }
            line.push(1);
            triangle.push(line);
        }
        triangle
    }
}

File: algorithms/0118.pascals_triangle/pascals_triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> ans;
        ans.push_back({1});
        for (int i = 1; i < numRows; ++i) {
            const auto& last = ans.back();
            auto cur = last;
            cur.push_back(1);
            for (int j = 2; j <= i; ++j) {
                cur[j - 1] = last[j - 2] + last[j - 1];
            }
            ans.emplace_back(cur);
        }
        return ans;
    }
};